lib/ProducerConfigurationImpl.h

```diff
@@ -50,6 +50,20 @@ struct ProducerConfigurationImpl {
     std::map<std::string, std::string> properties;
     bool chunkingEnabled{false};
     ProducerConfiguration::ProducerAccessMode accessMode{ProducerConfiguration::Shared};
+    std::string initialSubscriptionName;
+
+    /**
+     * Use this config to automatically create an initial subscription when creating the topic.
+     * If this field is not set, the initial subscription will not be created.
+     * This method is limited to internal use
+     *
+     * @param initialSubscriptionName Name of the initial subscription of the topic.
+     */
+    void setInitialSubscriptionName(const std::string& initialSubscriptionNameParam) {
+        initialSubscriptionName = initialSubscriptionNameParam;
+    }
+
+    const std::string& getInitialSubscriptionName() const { return initialSubscriptionName; }
```

Comment:
It's better not to add a setter and getter to a POD class. `initialSubscriptionName` is a public field, we can modify it directly. Adding a getter and setter to a POD class is redundant and adds much more code.